src/main/java/com/upserve/uppend/blobs/VirtualPageFile.java

```diff
@@ -482,13 +493,13 @@ private MappedByteBuffer ensureBuffered(int bufferIndex) {
     }
 
     private LongBuffer ensurePageTable(int pageNumber) {
-        LongBuffer buffer = pageTables[pageNumber];
+        MappedByteBuffer buffer = pageTables[pageNumber];
         if (buffer == null) {
             synchronized (pageTables) {
                 buffer = pageTables[pageNumber];
                 if (buffer == null) {
```

Comment:
These lines are confusing.

Reply:
Agreed - suggestions on how to improve?
You have to check the state of the array again after you enter the synchronized block.